tests/platform_tests/cli/test_show_platform_dpu.py

```diff
@@ -0,0 +1,183 @@
+    """
+    @summary: Verify `Link flap between NPU and DPU`
+    - Bringing all DPU interfaces down.
+    - Deleting the state table from redis db
+    - Checking show system-health dpu to see no output is there.
+      (Since the db entries are deleted and link down, state table would be empty which gets reflected in the cli)
+    - Brining up the interfaces.
+    - State table gets updated automatically after a minute
+    - Checking the show system-health dpu cli to get all the details
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    count_up = 0
+    count_down = 0
+    dpu_db_before = 0
+    dpu_db_after = 0
+
+    # Checking the state of dpu health before the flap
+    dpu_db_before = count_dpu_modules_in_system_health_cli(duthost)
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+
+        if interface != "eth0" and 'eth' in interface:
+            output_intf_down = duthost.shell("ifconfig %s down"%(interface))["stdout_lines"]
+            time.sleep(1)
+
+    # Deleting the DB table entry after bringing down the interface
+    duthost.shell(
+            "for key in `redis-cli -p 6380 -h 127.0.0.1 -n 13 --scan --pattern \"DPU_STATE|DPU*\" `; "
+            "do redis-cli -p 6380 -h 127.0.0.1 -n 13 del $key ; done")
+
+    # Checking the state of dpu health again after bringing down interface and deleting db entry
+    dpu_db = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
+
+    pytest_assert(dpu_db == 0, "Link is not down'{}'".format(duthost.hostname))
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+        status = parse_output['admin/oper']
+
+        if interface != "eth0" and 'eth' in interface:
+            if status == "down/down":
+                count_down += 1
+            output_intf_up = duthost.shell("ifconfig %s up"%(interface))["stdout_lines"]
+            time.sleep(1)
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+        status = parse_output['admin/oper']
+
+        if interface != "eth0" and 'eth' in interface:
+            if status == "up/up":
+                count_up += 1
+
+    # Checking the state of dpu health again after bringing up interface and waiting for db entry to get populated
+    dpu_db_after = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
+
+    pytest_assert(dpu_db_before == count_up == count_down == dpu_db_after, "Link Flap is Tested'{}'".format(duthost.hostname))
```

Comment:
It is good to have the DPU information whose link didn't come up after the test. Also, can we have a debug log in successful case?

Reply:
Added it.